### PR TITLE
feat: 新增流量统计模块

### PR DESCRIPTION
**Pull request by TanxiangCode**
增加了一个流量统计的功能

仅测试了win

### CODE REVIEW

File: src/services/traffic-worker.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  hour: { format: "YYYY-MM-DD HH:00", logPeriod: 24 },
+  day: { format: "YYYY-MM-DD", logPeriod: 31 },
+  month: { format: "YYYY-MM", logPeriod: 12 },
+  year: { format: "YYYY ", logPeriod: 1 },
+};
+const SupportedTimeUnits = Object.keys(StatisticConfig) as TimeUnitType[];
+
+let aggregationTimer: NodeJS.Timeout;
+
+self.onmessage = async (event) => {
```
Comment (SukkaW) [inline src/services/traffic-worker.ts:26]: Use `addEventListener`.

File: src/services/traffic-worker.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+  const { type, data } = event.data;
```
Comment (SukkaW) [inline src/services/traffic-worker.ts:27]: Missing typescript type declarations.

File: src/services/traffic-worker.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  };
+
+  // 初始化
+  await aggregate();
+}
+
+// 自启动后台任务
+(async () => {
+  await startBackgroundTask();
+})();
```
Comment (SukkaW) [inline src/services/traffic-worker.ts:170]:
```suggestion
startBackgroundTask();
```

Also missing dispose function.

File: src/components/layout/layout-traffic.tsx [hunk 4]
```diff
@@ -16,6 +16,15 @@ import { createSockette } from "@/utils/websocket";
 import { useTranslation } from "react-i18next";
 import { isDebugEnabled, gc } from "@/services/api";
 
+console.log(new URL("@/services/traffic-worker.ts", import.meta.url));
```
Comment (SukkaW) [inline src/components/layout/layout-traffic.tsx:19]: Remove `console.log`.

File: src/components/layout/layout-traffic.tsx [hunk 5]
```diff
@@ -16,6 +16,15 @@ import { createSockette } from "@/utils/websocket";
...
+const trafficWorker = new Worker(
+  new URL("@/services/traffic-worker.ts", import.meta.url),
+  { type: "module" }
+);
+trafficWorker.onerror = (event) => {
+  console.error(event);
+};
```
Comment (SukkaW) [inline src/components/layout/layout-traffic.tsx:26]: Use `addEventListener`. Also add proper error handling.

**Review by SukkaW — CHANGES_REQUESTED** [review]

**Comment by TanxiangCode** [comment]
This approach has an issue: when minimized to the system tray, the worker will stop, and data writing will also cease. Alternative solutions need to be considered for data writing and processing.

**Pull request closed without merging**